fix: change duration of snackbar from short to medium

encetroc opened the pull request:
Fixes #3838 

#### Description

- I changed the default duration of snackbar from short to meduim
- I updated the documentation to reflect the changes made

#### Scope

Patch: Bug Fix

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber/baseweb?pullRequest=3842) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/uber/baseweb?pullRequest=3842) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber/baseweb?pullRequest=3842) it.</sub>

chasestarr commented:
Thanks for looking into this. Based on our design guidelines snackbars with one line of text should use the short duration. I assume that will be the most common case so the default should remain as short. When working on a particular application I may know that my snackbars more often have two lines. As a developer experience improvement I should be able to supply the `SnackbarProvider` with a `defaultDuration` prop so that I can save some typing and only configure the outliers.

```
<SnackbarProvider />

// this queued snackbar should render for the short duration
enqueue({message: 'hello world'});
```

```
<SnackbarProvider defaultDuration={DURATION.medium} />

// this queued snackbar should render for the medium duration
enqueue({message: 'hello world'});
```

encetroc commented:
Thanks for your clarification as this is my first time trying to contribute to open source. I updated my PR and added the possibility to change the default duration using the provider.

chasestarr commented on src/snackbar/snackbar-context.js:
```diff
@@ -47,6 +47,7 @@ export default function SnackbarProvider({
   children,
   overrides = {},
   placement,
+  defaultDuration,
```
```suggestion
  defaultDuration = DURATION.short,
```

chasestarr replied:
This way it only applies short if `defaultDuration` is undefined

chasestarr commented on src/snackbar/snackbar-context.js:
```diff
@@ -58,7 +59,7 @@ export default function SnackbarProvider({
   const containerRef = React.useRef(null);
 
-  function enqueue(elementProps, duration = DURATION.short) {
+  function enqueue(elementProps, duration = defaultDuration || DURATION.short) {
```
```suggestion
  function enqueue(elementProps, duration = defaultDuration) {
```

chasestarr reviewed (commented):
Thanks for making this change!

chasestarr reviewed (approved):
Thank you
